apps/interactor/src/state/listeners.ts

```diff
@@ -11,15 +11,36 @@ import { RootState } from './store'
 import textCompletion from '../libs/textCompletion'
 import PromptBuilder from '../libs/memory/PromptBuilder'
 
+let lastTime = new Date().getTime()
+
 const interactionEffect = async (
   dispatch: Dispatch,
   state: RootState,
   servicesEndpoint: string
 ) => {
   try {
+    const response = await fetch(
+      `${servicesEndpoint}/text/strategy/${state.settings.model}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        cache:
+          new Date().getTime() - lastTime > 600000 ? 'reload' : 'force-cache',
+      }
+    )
+    lastTime = new Date().getTime()
```

Comment:
this should be updated only if `new Date().getTime() - lastTime > 60000`

```suggestion
    if (new Date().getTime() - lastTime > 60000) lastTime = new Date().getTime()
```

another easier thing you can do is to query only once in the roleplay and store it in a map. I think that might just work fine

Reply:
nevermind, we can remove `lastTime` as I mentioned in my other comment